src/jobs/jobs.controller.ts

```diff
@@ -475,18 +476,98 @@ export class JobsController {
     @Req() request: Request,
     @Body() createJobDto: CreateJobDto,
   ): Promise<JobClass> {
-    Logger.log("creating job!")
+    Logger.log("Creating job!");
     // Validate that request matches the current configuration
-    await this.validateJob(createJobDto, request);
+    await this.validateJob(createJobDto);
     // Check job authorization
-    await this.instanceAuthorization(createJobDto, request.user as JWTUser);
+    await this.instanceAuthorization(createJobDto.jobParams, request.user as JWTUser);
     // Create actual job in database
     const createdJobInstance = await this.jobsService.create(createJobDto);
     // Perform the action that is specified in the create portion of the job configuration
-    const jobServiceResponse = await this.performJobCreateAction(createdJobInstance);
+    await this.performJobCreateAction(createdJobInstance);
+    // TBD
     // Update job instance with results of job create action
-    // return await this.jobsService.update(jobServiceResponse);
-    return jobServiceResponse; // or createdJobInstance ?
+    // return await this.jobsService.statusUpdate(createdJobInstance);
+    return createdJobInstance;
+  }
+
+
+  /**
+   * Update job
+   */
+  @UseGuards(PoliciesGuard)
+  @CheckPolicies((ability: AppAbility) =>
+    ability.can(AuthOp.JobsUpdate, JobClass),
+  )
+  @Patch(":id")
+  @ApiOperation({
+    summary: "It updates an existing job.",
+    description: "It updates an existing job.",
+  })
+  @ApiBody({
+    description: "Input fields for the job to be updated",
+    required: true,
+    // schema: new UpdateJobStatusDto, // TBD
```

Comment:
This part is exactly the same as in `create()`. However, in `update()` it produces the following syntax error (hence why the line is commented out):
`Type 'UpdateJobStatusDto' is not assignable to type 'SchemaObject | ReferenceOb-ject'.`

What is the difference between the two DTOs? Why `UpdateJobStatusDto` is not accepted here?

Reply:
I looked into this issue. I think we are actually using this wrong everywhere, but coincidentally it compiles everywhere but here.

The ApiBody is expecting a JSON Schema description. The type definition is in `@nestjs/swagger/dist/interfaces/open-api-spec.interface.d.ts`. All fields are optional. Typescript enforces that the class passed have at least one of these properties and ignores additional properties. In the case of `CreateJobDto` (as well as `CreateDatasetDto`) there is a `type: string` parameter that happens to be compatible with the expected type. However, I guess that DTO validation isn't actually working in BE-next, and we just didn't notice until now.

Reply:
Testing to be covered by @rkweehinzmann, as mentioned in #626